src/settings.js

```diff
@@ -1630,6 +1630,12 @@ var ABORT_ON_WASM_EXCEPTIONS = 0;
 // Implies STANDALONE_WASM.
 var PURE_WASI = 0;
 
+// Extra items to provide as imports to the wasm. This is a list of items, each
+// of which can be a string X, in which case we import the global JavaScript
+// name X as "X", or a pair [KEY, VALUE], in which case we import the name
+// VALUE as "KEY".
+var EXTRA_WASM_IMPORTS = [];
```

Comment:
I think I'd rather see this just support a strings.. since those are the only things that can be reasonably set via the command line.  

It seems like only these 3 items would need to be changed:

```
        ('__memory_base', str(shared.Settings.GLOBAL_BASE)),
￼        ('__indirect_function_table', 'wasmTable'),
￼        # the backend reserves slot 0 for the NULL function pointer
￼        ('__table_base', '1'),
```

Trying to support pairs adds a fair bit of complexity.   It would be a tiny cod size regressions but only for MAIN_MODULE which already huge.

Reply:
I'm not sure what you're suggesting. Are you saying that for e.g. `__indirect_function_table` we would, in another file, need to do
```javascript
var __indirect_function_table = wasmTable;
```
And then for `EXTRA_WASM_EXPORTS` we can just write `__indirect_function_table` as then the import and the implementation names are the same?

If that's the suggestion, then I think that's more complicated and more verbose. It's nice to do it all on one line in the same file, as opposed to 2 files.

But you mention code size regressions so I'm not sure if I understood the suggestion?

Reply:
I think we should look at it on a case by case basis.  In general we should not encourage things to be exported under a different name since it lead to two different names for the same thing.

How about this:  Limit this change to only things that are exported under their own name.. thus simplify the change quite a bit but meaning that things like `__indirect_function_table` still need special handling?    This will discourage future folks from exporting things under different names.

Reply:
One general issue is that we do have the `_` prefixing, so the import `foo` is done with the JS function `_foo`. I think it's useful to allow that with this new option?

Reply:
How about this:  A public setting EXTRA_WASM_EXPORTS and an internal setting CUSTOM_WASM_EXPORTS which is for the custom pairs (where the internal and external name don't match).    That way we don't expose end users to setting that contains "pairs" and the `-s` command line syntax just only has to to worry about lists of strings.   And you don't need be doing dynamic typechecks on the contents of the `EXTRA_WASM_EXPORTS` list.

Reply:
Or at least don't document the fact that EXTRA_WASM_EXPORTS supports pairs.. or make it clear that pairs are and internal-only thing.

Reply:
I feel like all those add complexity. But maybe I don't understand what complexity you see them as removing.

Anyhow, I'm not sure this PR is still needed, given activity in the other PRs. So no rush to do anything here I guess.